godot-ffi/src/linux.rs

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! Linux-specific configuration.
+
+// Avoid TLS-destructors preventing the dynamic library from being closed.
+// See: https://fasterthanli.me/articles/so-you-want-to-live-reload-rust#what-can-prevent-dlclose-from-unloading-a-library
+
+use std::ffi::c_void;
+use std::sync::OnceLock;
+
+pub type NextFn = unsafe extern "C" fn(*mut c_void, *mut c_void, *mut c_void);
+
+static SYSTEM_THREAD_ATEXIT: OnceLock<Option<NextFn>> = OnceLock::new();
+static HOT_RELOADING_ENABLED: OnceLock<bool> = OnceLock::new();
+
+fn system_thread_atexit() -> &'static Option<NextFn> {
+    SYSTEM_THREAD_ATEXIT.get_or_init(|| unsafe {
+        #[allow(clippy::transmute_ptr_to_ref)]
+        let name = c"__cxa_thread_atexit_impl".as_ptr();
+        std::mem::transmute(libc::dlsym(
+            libc::RTLD_NEXT,
+            #[allow(clippy::transmute_ptr_to_ref)]
+            name,
+        ))
+    })
+}
+
+pub fn enable_hot_reload() {
+    // If hot reloading is enabled then we should properly unload the library, so this will only be called once.
+    HOT_RELOADING_ENABLED
+        .set(true)
+        .expect("hot reloading should only be set once")
+}
+
+pub fn disable_hot_reload() {
+    // If hot reloading is disabled then we may call this method multiple times.
+    _ = HOT_RELOADING_ENABLED.set(false)
+}
+
+pub fn default_set_hot_reload() {
+    // By default we enable hot reloading for debug builds, as it's likely that the user may want hot reloading in debug builds.
+    // Release builds however should avoid leaking memory, so we disable hot reloading support by default.
+    if cfg!(debug_assertions) {
+        enable_hot_reload()
+    } else {
+        disable_hot_reload()
+    }
+}
```

Comment:
Any ideas on this? 🙂

Reply:
maybe by default? im not sure how to do that since we need to make sure to run this code very early on in the initialization and im not sure if we can check that quite that early on